File: LeetCode/Easy/CPP/206ReverseLinkedList.cpp
#include <bits/stdc++.h>

using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *temp;
        ListNode *ans = nullptr;

        temp = head;
        while (temp)
        {
            ListNode node(temp->val);
            node.next = ans;
            ans = &node;
            temp = temp->next;
        }
        return ans;
    }
};

int main(void)
{
    ListNode list(5);
    ListNode *temp;

    temp = &list;
    ListNode list2(9);
    list2.next = temp;
    temp = &list2;
    while (temp)
    {
        cout << temp->val << endl;
        temp = temp->next;
    }
    return(0);
}

File: LeetCode/Easy/PYTHON/206ReverseLinkedList.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev_elem = None
        curr_elem = head
        next_elem = None

        if head == None or head.next == None:
            return head
        
        while curr_elem != None:
            next_elem = curr_elem.next
            curr_elem.next = prev_elem
            prev_elem = curr_elem
            curr_elem = next_elem
        return prev_elem